Rework mspi device binding and compat strings. (#108082)
zephyrproject-rtos/zephyr: samples/drivers/mspi/mspi_async/README.rst

The application will build only for a target that has a :ref:`devicetree <dt-guide>`

> ``dev0`` alias that refers to an entry with the following bindings as a compatible:

Reviewer: ```suggestion
``dev0`` alias that refers to an entry with the following compatible strings:
```
